src/com/palmergames/bukkit/towny/command/ResidentCommand.java

```diff
@@ -132,6 +140,8 @@ public boolean onCommand(CommandSender sender, Command cmd, String commandLabel,
 				case "toggle":
 					if (args.length == 2) {
 						return NameUtil.filterByStart(residentToggleTabCompletes, args[1]);
+					} else if (args.length == 3 && residentToggleChoices.contains(args[1].toLowerCase())) {
+						return NameUtil.filterByStart(BaseCommand.setOnOffCompletes, args[2]);

```

Comment:
This new residentToggleChoices option isn't needed. All of the existing residentToggleTabCompletes are things that can (and probably should) toggled on/off with the new discrete toggling.

Reply:
I did this change because for the ones who aren't in the `residentToggleChoices` array, we expect a "mode" value, not an on/off value; this is displayed here:
https://github.com/FrankHeijden/Towny/blob/feature/discrete-toggles/src/com/palmergames/bukkit/towny/command/ResidentCommand.java#L427

Which points to here:
https://github.com/FrankHeijden/Towny/blob/feature/discrete-toggles/src/com/palmergames/bukkit/towny/object/Resident.java#L537

In that last method, we don't want discrete toggles, we want the modes there.

Reply:
Yeah it's a bit confused by the way it was originally done. Before there was any `/res toggle` all the modes were set using `/res set mode` but later on (since all modes are just on/offs) they were moved to being accessible via `/res toggle' (while still having the original `/res set mode`.)

Ideally all of the modes would have on|off as options through `/res toggle` but it might be a bit more work to accomplish this. (ex `/res toggle townclaim on|off`)

Reply:
Done! Sorry for the holdup, the behaviour for this section is now a bit more dynamic:
`/res toggle townclaim`
`/res toggle townclaim on|off ignoreplots`
`/res toggle townclaim ignoreplots on|off`

All of the above commands are valid, if the user didn't specify on|off after a mode, the mode will be toggled (old behaviour).